src/App.js

```diff
@@ -1,23 +1,37 @@
-import logo from './logo.svg';
+import { HashRouter as Router, Routes, Route } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { React, useEffect } from 'react';
+import { fetchData, randOpacity } from './redux/home/home';
+import Header from './components/header';
+import Home from './pages/home';
+import Continent from './pages/continent';
+import Details from './pages/details';
 import './App.css';
 
 function App() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchData());
+  }, []);
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <Router>
+        <Header />
+        <Routes>
+          <Route exact path="/" element={<Home randOpacity={randOpacity} />} />
```

Comment:
- [ ] The Home page should have list of categories that could be filtered using given parameters, e.g. country name. I'd suggest creating stateless controlled component for filtering the content based on a text query eg.  country name.

Reply:
Can you explain what stateless controlled components are to me? I couldn't figure out what was meant with them, so currently I'm filtering by continents only, but if you could explain, I would be very grateful.